configuration.py

```diff
@@ -0,0 +1,153 @@
+import yaml
+import datetime
+import calendar
+import re
+from sharepoint_utils import print_and_log
+
+
+class Configuration:
+    def __init__(self, file_name, logger=None):
+        self.logger = logger
+        try:
+            with open(file_name, 'r') as stream:
+                self.configurations = yaml.safe_load(stream)
+        except Exception as exception:
+            print_and_log(self.logger, 'error', "Error while reading the configurations from {}. Error: {}".format(
+                file_name, exception))
+
+    def validate_date(self, param_name, input_date):
+        current_time = int(calendar.timegm(
+            datetime.datetime.utcnow().timetuple()))
+        if input_date:
+            if not re.match(r"^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}$", input_date):
+                self.logger.warn(
+                    "The parameter {} is not in the right format. Expected format \"YYYY-MM-DDThh:mm:ss\", found: {}".format(param_name, input_date))
+                return False
+
+            try:
+                input_date = int(calendar.timegm(datetime.datetime.strptime(
+                    input_date, "%Y-%m-%dT%H:%M:%S").timetuple()))
+            except Exception:
+                self.logger.warn(
+                    "The parameter {} is not in the right format. Expected format \"YYYY-MM-DDThh:mm:ss\", found: {}".format(param_name, input_date))
```

Comment:
you are assuming this but your oneline is doing 3 operations. This warning can be misleading. You should first parse the date then do the conversions outside this block

Reply:
Done